src/Entity/Donation.php

```diff
@@ -242,4 +258,31 @@ public function getRetryPayload(): array
 
         return $payload;
     }
+
+    public function getSubscriptionEndedAt(): ?\DateTime
+    {
+        return $this->subscriptionEndedAt;
+    }
+
+    public function setSubscriptionEndedAt(?\DateTime $subscriptionEndedAt): void
+    {
+        $this->subscriptionEndedAt = $subscriptionEndedAt;
+    }
+
+    public function nextDonationAt(): \DateTime
+    {
+        $day = $this->donatedAt->format('d');
+        $dayNow = date('d');
+        if ($day > $dayNow) {
```

Comment:
how it's possible?

Reply:
now is 5/02/2015 , the subscription have been done in 15/01/2015 then the next date will be 15/02/2015
 
if now is 20/02/2015 next date will be 15/03/2015

Reply:
how it's possible that the donation date is in the future?

Reply:
It's for subscription donation